### PR TITLE
Fixing major issues

### PR DESCRIPTION
**Pull request by neilprado**

### CODE REVIEW

File: heimdall-core/src/main/java/br/com/conductor/heimdall/core/converter/GenericConverter.java [hunk 1]
```diff
@@ -36,6 +36,10 @@
  */
 public abstract class GenericConverter {
 
+     private GenericConverter(){
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-core/src/main/java/br/com/conductor/heimdall/core/converter/GenericConverter.java:39]: Reverted

File: heimdall-api/src/main/java/br/com/conductor/heimdall/api/util/ZipUtils.java [hunk 2]
```diff
@@ -64,7 +67,9 @@ private ZipUtils() { }
             zipOutputStream.flush();
             return byteArrayOutputStream.toByteArray();
 
-        } catch (IOException ignored) { }
+        } catch (IOException exception) {
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-api/src/main/java/br/com/conductor/heimdall/api/util/ZipUtils.java]: Reverted

File: heimdall-core/src/main/java/br/com/conductor/heimdall/core/dto/metrics/Metric.java [hunk 3]
```diff
@@ -32,9 +32,9 @@
 public class Metric {
 
     @JsonIgnore
-    private String _id;
+    private String id;
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-core/src/main/java/br/com/conductor/heimdall/core/dto/metrics/Metric.java]: Fixed

File: heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/DBMockImpl.java [hunk 4]
```diff
@@ -170,7 +173,8 @@
 
             Document document = Document.parse(json.parse(object));
             collection.insertOne(document);
-        } catch (Exception ignored) {
+        } catch (Exception exception) {
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/DBMockImpl.java]: Reverted

File: heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/DBMockImpl.java [hunk 5]
```diff
@@ -291,7 +295,9 @@ private MongoDatabase database() {
         id.setAccessible(true);
         try {
             return id.get(object);
-        } catch (IllegalArgumentException | IllegalAccessException ignored) { }
+        } catch (IllegalArgumentException | IllegalAccessException exception) {
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/DBMockImpl.java]: Reverted

File: heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/DBMockImpl.java [hunk 6]
```diff
@@ -305,7 +311,9 @@ private MongoDatabase database() {
             Object value = null;
             try {
                 value = field.get(criteria);
-            } catch (IllegalArgumentException | IllegalAccessException ignored) {}
+            } catch (IllegalArgumentException | IllegalAccessException exception){
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/DBMockImpl.java]: Reverted

File: heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/DBMockImpl.java [hunk 7]
```diff
@@ -51,15 +52,16 @@
  *
  * @author Marcelo Aguiar
  */
+@Slf4j
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/DBMockImpl.java]: Annotation removed

File: heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/JsonMockImpl.java [hunk 8]
```diff
@@ -137,15 +139,17 @@ public boolean isJson(String string) {
...
 
             valid = true;
         } catch (JSONException e) {
 
             try {
                 new JSONArray(string);
                 valid = true;
-            } catch (JSONException ignored) { }
+            } catch (JSONException exception) {
+                log.error(exception.getMessage(), exception);
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/JsonMockImpl.java:151]: Reverted

File: heimdall-api/src/main/java/br/com/conductor/heimdall/api/resource/TracesResource.java [hunk 9]
```diff
@@ -74,7 +74,9 @@
     @PreAuthorize(ConstantsPrivilege.PRIVILEGE_READ_TRACES)
     public ResponseEntity<?> findOne(@PathVariable("traceId") String id) {
     	
-    	if (!property.getMongo().getEnabled()) return ResponseEntity.ok(new JSONObject().toString());
+    	if (!property.getMongo().getEnabled()){
+    	    return ResponseEntity.ok(new JSONObject().toString());
+        }
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-api/src/main/java/br/com/conductor/heimdall/api/resource/TracesResource.java]: Brackets removed

File: heimdall-api/src/main/java/br/com/conductor/heimdall/api/resource/TracesResource.java [hunk 10]
```diff
@@ -94,7 +96,9 @@
     @PreAuthorize(ConstantsPrivilege.PRIVILEGE_READ_TRACES)
     public ResponseEntity<?> find(@RequestBody List<FiltersDTO> filtersSelected, @ModelAttribute PageableDTO pageableDTO) {
     	
-    	if (!property.getMongo().getEnabled()) return ResponseEntity.ok(new JSONObject().toString());
+    	if (!property.getMongo().getEnabled()){
+    	    return ResponseEntity.ok(new JSONObject().toString());
+        }
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-api/src/main/java/br/com/conductor/heimdall/api/resource/TracesResource.java]: Brackets removed

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/appender/MongoDBAppender.java [hunk 11]
```diff
@@ -111,10 +111,10 @@ public void stop() {
 
 	@Override
 	protected void append(ILoggingEvent e) {
-        ZoneId zoneId = ZoneId.of(this.zoneId);
+        ZoneId zone = ZoneId.of(this.zoneId);
```
[the remark this answers is not in the document]
Comment (neilprado) [reply heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/appender/MongoDBAppender.java:114]: Done

File: heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/JsonMockImpl.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (smrafael) [reply heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/JsonMockImpl.java:151]: You forgot to revert

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/appender/MongoDBAppender.java [hunk 12]
```diff
@@ -114,7 +114,7 @@ protected void append(ILoggingEvent e) {
         ZoneId zoneId = ZoneId.of(this.zoneId);
 
         // Offset in milliseconds based on the informed Zone
-        long offset = zoneId.getRules().getOffset(Instant.now()).getTotalSeconds() * 1000;
+        long offset = (long) zoneId.getRules().getOffset(Instant.now()).getTotalSeconds() * 1000;
```
Comment (smrafael) [inline heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/appender/MongoDBAppender.java:117]: Cast is really necessary?

File: heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/JsonMockImpl.java [hunk 13]
```diff
@@ -41,6 +42,7 @@
  *
  * @author Marcelo Aguiar
  */
+@Slf4j
```
Comment (smrafael) [inline heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/JsonMockImpl.java:45]: Remove logger

Comment (neilprado) [reply heimdall-middleware-spec/src/main/java/br/com/conductor/heimdall/middleware/util/helpermock/JsonMockImpl.java:45]: Done

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/appender/MongoDBAppender.java (hunk 12 above)
Comment (neilprado) [reply heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/appender/MongoDBAppender.java:117]: Cast removed. Offset variable now is int.

**Pull request merged**